src/include/mallocMC/creationPolicies/Scatter_impl.hpp

```diff
@@ -744,7 +740,7 @@ namespace ScatterKernelDetail{
         uint32* regions = (uint32*)(ptes + numpages);
         //sec check for mem size
         //this check refers to the original memory-pointer, which was not adjusted!
-        if( (void*)(regions + numregions) > (((char*)memory) + memsize) )
+        if( (void*)(regions + numregions) > (((char*)memory) + memsize) ) // TODO(bgruber): cast to void* is confusing and looks dangerous
```

Comment:
Feel free to cast the left hand side to `char*` too

Reply:
done